### ISSUE TITLE
DeepSpeed during backward() RuntimeError: ProcessGroup nccldoes not support _reduce_scatter_base

### ISSUE THREAD

**Issue by memray** [huggingface/accelerate#1098]
### System Info

```Shell
- `Accelerate` version: 0.16.0
- Platform: Linux-5.10.133+-x86_64-with-glibc2.17
- Python version: 3.8.12
- Numpy version: 1.22.2
- PyTorch version (GPU?): 1.11.0+cu113 (True)
- `Accelerate` default config:
	- compute_environment: LOCAL_MACHINE
	- distributed_type: DEEPSPEED
	- mixed_precision: bf16
	- use_cpu: False
	- dynamo_backend: NO
	- num_processes: 1
	- machine_rank: 0
	- num_machines: 1
	- rdzv_backend: static
	- same_network: True
	- main_training_function: main
	- deepspeed_config: {'gradient_accumulation_steps': 1, 'gradient_clipping': 1.0, 'offload_optimizer_device': 'cpu', 'offload_param_device': 'cpu', 'zero3_init_flag': True, 'zero_stage': 3}
	- fsdp_config: {}
	- megatron_lm_config: {}
	- downcast_bf16: no
```

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [ ] My own task or dataset (give details below)

### Reproduction

Can you confirm if this is a DeepSpeed bug? I can ask over there.
Trained on an A100*16 instance, failed with either 1gpu or 16gpus.
Training with stage2 is normal.

Code: https://github.com/huggingface/accelerate/blob/main/examples/by_feature/deepspeed_with_config_support.py

Command:
CUDA_VISIBLE_DEVICES=0 accelerate launch --config_file gpu1-Stage3-config.yaml deepspeed_with_config_support.py --model_name_or_path salesforce/codegen-2B-multi --dataset_name wikitext --dataset_config_name wikitext-103-v1 --per_device_train_batch_size 1 --per_device_eval_batch_size 2 --output_dir output/FT-mktcloud-test --max_train_steps 20 --num_warmup_steps 5 --with_tracking --learning_rate 1e-5

```
Traceback (most recent call last):
  File "deepspeed_with_config_support.py", line 732, in <module>
    main()
  File "deepspeed_with_config_support.py", line 641, in main
    accelerator.backward(loss)
  File "/export/share/ruimeng/env/anaconda/envs/codegen/lib/python3.8/site-packages/accelerate/accelerator.py", line 1435, in backward
    self.deepspeed_engine_wrapped.backward(loss, **kwargs)
  File "/export/share/ruimeng/env/anaconda/envs/codegen/lib/python3.8/site-packages/accelerate/utils/deepspeed.py", line 168, in backward
    self.engine.backward(loss)
......
  File "/export/share/ruimeng/env/anaconda/envs/codegen/lib/python3.8/site-packages/deepspeed/comm/comm.py", line 268, in reduce_scatter_fn
    return reduce_scatter_base(output_tensor,
  File "/export/share/ruimeng/env/anaconda/envs/codegen/lib/python3.8/site-packages/deepspeed/comm/comm.py", line 127, in log_wrapper
    return func(*args, **kwargs)
  File "/export/share/ruimeng/env/anaconda/envs/codegen/lib/python3.8/site-packages/deepspeed/comm/comm.py", line 302, in reduce_scatter_base
    return cdb.reduce_scatter_base(output_tensor=output_tensor,
  File "/export/share/ruimeng/env/anaconda/envs/codegen/lib/python3.8/site-packages/deepspeed/comm/torch.py", line 102, in reduce_scatter_base
    return torch.distributed._reduce_scatter_base(output_tensor,
  File "/export/share/ruimeng/env/anaconda/envs/codegen/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py", line 2484, in _reduce_scatter_base
    work = group._reduce_scatter_base(output, input, opts)
RuntimeError: ProcessGroup nccldoes not support _reduce_scatter_base```


│ /export/share/ruimeng/env/anaconda/envs/codegen/lib/python3.8/site-packages/ │
│ deepspeed/comm/comm.py:302 in reduce_scatter_base                            │
│                                                                              │
│   299 │   │   │   │   │   │   log_name='reduce_scatter_base',                │
│   300 │   │   │   │   │   │   debug=get_caller_func()):                      │
│   301 │   global cdb                                                         │
│ ❱ 302 │   return cdb.reduce_scatter_base(output_tensor=output_tensor,        │
│   303 │   │   │   │   │   │   │   │      input_tensor=tensor,                │
│   304 │   │   │   │   │   │   │   │      op=op,                              │
│   305 │   │   │   │   │   │   │   │      group=group,                        │
│                                                                              │
│ /export/share/ruimeng/env/anaconda/envs/codegen/lib/python3.8/site-packages/ │
│ deepspeed/comm/torch.py:102 in reduce_scatter_base                           │
│                                                                              │
│    99 │   │   │   │   │   │   │   group=None,                                │
│   100 │   │   │   │   │   │   │   async_op=False):                           │
│   101 │   │   if self.has_reduce_scatter_base:                               │
│ ❱ 102 │   │   │   return torch.distributed._reduce_scatter_base(output_tenso │
│   103 │   │   │   │   │   │   │   │   │   │   │   │   │   │     input_tensor │
│   104 │   │   │   │   │   │   │   │   │   │   │   │   │   │     op=self._red │
│   105 │   │   │   │   │   │   │   │   │   │   │   │   │   │     group=group, │
│                                                                              │
│ /export/share/ruimeng/env/anaconda/envs/codegen/lib/python3.8/site-packages/ │
│ torch/distributed/distributed_c10d.py:2484 in _reduce_scatter_base           │
│                                                                              │
│   2481 │   │   default_pg = _get_default_group()                             │
│   2482 │   │   work = default_pg._reduce_scatter_base(output, input, opts)   │
│   2483 │   else:                                                             │
│ ❱ 2484 │   │   work = group._reduce_scatter_base(output, input, opts)        │
│   2485 │                                                                     │
│   2486 │   if async_op:                                                      │
│   2487 │   │   return work                                                   │
╰──────────────────────────────────────────────────────────────────────────────╯
RuntimeError: ProcessGroup nccldoes not support _reduce_scatter_base
```

### Expected behavior

```Shell
No error.
```

**Comment by sgugger** [comment]
cc @pacman100

**Comment by memray** [comment]
Problem resolved after upgrading to deepspeed==0.8.1 and torch==1.12.1+cu113
